Correct return type of strlen function.

Damien-Chen opened the pull request:
I am confused about the return type of strlen() function.
In general, strlen() returns the length of the input string, which is always non‑negative.
Switching the return type to uint32_t is more accurately.

visitorckw commented:
Thanks for the patch!
Just a couple of suggestions: the subject line of the commit message shouldn't end with a period. Also, instead of changing the return type to uint32_t, it might be more appropriate to use size_t. While Linmo currently only supports RV32 systems, using size_t makes the code more portable and could matter if we support RV64 in the future.

Damien-Chen commented:
Hi @visitorckw,
Thanks for your feedback, I have updated new patch and squashed it into single one commit.
If there is anything need to improve, please let me know.

visitorckw reviewed (approved):
LGTM.
